flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -1373,6 +1380,12 @@ private void disconnectAndTryReconnectToJobManager(
         }
     }
 
+    private void disconnectAndTryReconnectToJobManagerAndCleanupPartitionLater(
+            JobTable.Connection jobManagerConnection, Exception cause) {
+        disconnectJobManagerConnectionAndCleanupPartitionLater(jobManagerConnection, cause);
```

Comment:
I prefer to name it as `disconnectAndTryReconnectToJobManager` and add comments to explain that it does not cleanup partitions right now. Only if the reconnection cannot be done will the partitions get cleared.

It is natural in the case to support JM reconnection so we do not need add all the details to the method name. And `CleanupPartitionLater` may not happen.